src/dappsExplorer/DAppsExplorerScreenFilter.test.tsx

```diff
@@ -0,0 +1,426 @@
+        </Provider>
+      )
+
+      // don't include events dispatched on screen load
+      jest.clearAllMocks()
+
+      // Tap on category 2 filter
+      fireEvent.press(getByTestId('DAppsExplorerScreenFilter/FilterChip/2'))
+
+      expect(ValoraAnalytics.track).toHaveBeenCalledTimes(1)
+      expect(ValoraAnalytics.track).toHaveBeenCalledWith(DappExplorerEvents.dapp_filter, {
+        id: '2',
+        remove: false,
+      })
+    })
+
+    it('triggers event when clearing filters from all section', () => {
+      const store = createMockStore({
+        dapps: {
+          dappListApiUrl: 'http://url.com',
+          dappsList,
+          dappsCategories,
+          dappFavoritesEnabled: true,
+          dappsFilterEnabled: true,
+          favoriteDappIds: ['dapp2'],
+        },
+      })
+      const { getByTestId } = render(
+        <Provider store={store}>
+          <DAppsExplorerScreenFilter />
+        </Provider>
+      )
+
+      // don't include events dispatched on screen load
+      jest.clearAllMocks()
+
+      // Tap on category 2 filter
+      fireEvent.press(getByTestId('DAppsExplorerScreenFilter/FilterChip/2'))
+
+      // Tap on remove filters from all section
+      fireEvent.press(getByTestId('DAppsExplorerScreenFilter/NoResults/RemoveFilter'))
+
+      // Assert correct analytics are fired
+      expect(ValoraAnalytics.track).toHaveBeenCalledTimes(2)
+      expect(ValoraAnalytics.track).toHaveBeenCalledWith(DappExplorerEvents.dapp_filter, {
+        id: '2',
+        remove: false,
+      })
+      expect(ValoraAnalytics.track).toHaveBeenCalledWith(DappExplorerEvents.dapp_filter, {
+        id: '2',
+        remove: true,
+      })
+    })
+
+    it('triggers event when clearing filters from favorite section', () => {
+      const store = createMockStore({
+        dapps: {
+          dappListApiUrl: 'http://url.com',
+          dappsList,
+          dappsCategories,
+          dappFavoritesEnabled: true,
+          dappsFilterEnabled: true,
+          favoriteDappIds: ['dapp1'],
+        },
+      })
+      const { getByTestId } = render(
+        <Provider store={store}>
+          <DAppsExplorerScreenFilter />
+        </Provider>
+      )
+
+      // don't include events dispatched on screen load
+      jest.clearAllMocks()
+
+      // Tap on category 2 filter
+      fireEvent.press(getByTestId('DAppsExplorerScreenFilter/FilterChip/2'))
+
+      // Tap on remove filters from all section
+      fireEvent.press(getByTestId('FavoriteDappsSection/NoResults/RemoveFilter'))
+
+      // Assert correct analytics are fired
+      expect(ValoraAnalytics.track).toHaveBeenCalledTimes(2)
+      expect(ValoraAnalytics.track).toHaveBeenCalledWith(DappExplorerEvents.dapp_filter, {
+        id: '2',
+        remove: false,
+      })
+      expect(ValoraAnalytics.track).toHaveBeenCalledWith(DappExplorerEvents.dapp_filter, {
+        id: '2',
+        remove: true,
+      })
+    })
+  })
+})
```

Comment:
nice filters tests :) fwiw i think we should prefer `getByText` over `getByTestId` where it's possible. i started thinking this after reading this [article](https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#not-querying-by-text) but also i think the way that we gain the most confidence is seeing our users use the feature. the closer we can get to testing what a real user sees, the more confidence we'd gain

Reply:
Fixed in cbac8b8506318a72cec64093ec4bcb2164fdc7a4!